src/sql/optimizer/dynamic_partition_pruning.rs

```diff
@@ -781,6 +782,7 @@ fn satisfies_int64(long_value: Option<i64>, filter: Expr) -> bool {
                 Expr::Literal(ScalarValue::Int32(i)) => i64::from(i.unwrap()),
                 Expr::Literal(ScalarValue::Float64(i)) => i.unwrap() as i64,
                 Expr::Literal(ScalarValue::TimestampNanosecond(i, None)) => i.unwrap(),
+                Expr::Literal(ScalarValue::Date32(i)) => i64::from(i.unwrap()),
```

Comment:
Is this something you think we would want to apply for the other `satisfies_*` functions as well?

Reply:
Sure, I don't think it would hurt to add `Expr::Literal(ScalarValue::Utf8(_)) => return false,` to the equivalent places in `satisfies_int32` and `satisfies_float`.